### PR TITLE
[ISSUE #661] Message sending supports the specified queue

### PR DESCRIPTION
**Pull request by Git-Yang**
#661 
## What is the purpose of the change
- 支持指定queue发送单条/批量消息

## Brief changelog
- 增加支持queue发送的接口 
- 增加支持queue发送接口单元测试

## Verifying this change

XXXX

Follow this checklist to help us incorporate your contribution quickly and easily. Notice, `it would be helpful if you could finish the following 5 checklist(the last one is not necessary)before request the community to review your PR`.

- [x] Make sure there is a [Github issue](https://github.com/apache/rocketmq/issues) filed for the change (usually before you start working on it). Trivial changes like typos do not require a Github issue. Your pull request should address just this issue, without pulling in other changes - one PR resolves one issue. 
- [x] Format the pull request title like `[ISSUE #123] Fix UnknownException when host config not exist`. Each commit in the pull request should have a meaningful subject line and body.
- [x] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
- [x] Write necessary unit-test(over 80% coverage) to verify your logic correction, more mock a little better when a cross-module dependency exists.
- [ ] If this contribution is large, please file an [Apache Individual Contributor License Agreement](http://www.apache.org/licenses/#clas).

### CODE REVIEW

**Comment by wenfengwang** [comment]
@Git-Yang Hi Yang, there is a conflict, could you resolve it?

**Comment by Git-Yang** [comment]
> @Git-Yang Hi Yang, there is a conflict, could you resolve it?

Please review it for me

**Review by maixiaohai — APPROVED** [review]

File: producer/producer.go [hunk 1]
```diff
@@ -163,35 +168,77 @@ func (p *defaultProducer) SendSync(ctx context.Context, msgs ...*primitive.Messa
 			var err error
 			realReq := req.(*primitive.Message)
 			realReply := reply.(*primitive.SendResult)
-			err = p.sendSync(ctx, realReq, realReply)
+			return p.sendSyncWithQueue(ctx, mq, realReq, realReply)
 			return err
```
Comment (wenfengwang) [inline producer/producer.go:172]: remove this line?

**Review by wenfengwang — CHANGES_REQUESTED** [review]

File: producer/producer.go (hunk 1 above)
Comment (Git-Yang) [reply producer/producer.go:172]: Why remove this line?

**Comment by Git-Yang** [comment]
@wenfengwang Please review again, thanks!

**Comment by Git-Yang** [comment]
@vongosling Can you please review the code for me ？

**Pull request closed without merging**